dayOne/PrimeNumber.java

```diff
@@ -0,0 +1,16 @@
+package dayOne;
+import java.util.Scanner;
+
+public class PrimeNumber {
+	public static void main(String[] args) {
+		Scanner scan=new Scanner(System.in);
+		System.out.println("Enter a no:");
+		int no=scan.nextInt();
+		boolean flag=true;
+		for(int i=2;i<=no/2;i++)
```

Comment:
Your solution is correct. Is there any way to optimize iterations even further?

Reply:
This loop will check all the numbers between 2 and number/2. Checking 2 and odd numbers separately will be an optimised way.

if(no%2==0)return false;

for(int i=3; I<=no/2; i+=2)
if(no%i==0)
return false;

Reply:
Imagine the inputNumber is huge, say 15000, even with inputNumber/2 condition, there'll be a lot of iterations(7500). Instead, iterating till the square root of the inputNumber (123 iterations) would itself determine whether or not it's a prime number. There are ways to even skip the 123 iterations by having some basic if conditions at the start. 
`If(inputNumber == 2 || inputNumber == 3)
    return true`
`if(inputNumber%2==0 || inputNumber%3==0)
   return false
`
`
for(int value=5;value * value <= inputNumber; value+=2)
if(inputNumber % value == 0)
return false
`

Reply:
Understood, sir. I will change it.